File: bin/printScoreFile_byHeader.py
#! python3
import sys, os

def main():
    if len(sys.argv) < 3:
        print('The script needs a Rosetta scorefile and at least one name of a score to output. Exiting...')
        exit()

    score_file = sys.argv[1]
    scores = sys.argv[2:]
    src_file = os.path.split(score_file)[1]

    # process line-by line, skip lines with SEQUENCE
    print('src_file' + ' ' + " ".join(scores))
    with open(score_file, 'r') as f:
        header = []
        indices = []
        for line in f:
            if line.startswith('SEQUENCE'):
                continue
            if line.startswith('SCORE'):
                if len(header) == 0:
                    header = line.strip().split()
                    for score in scores:
                        try:
                            idx = header.index(score)
                            indices.append(idx)
                        except:
                            continue
                else:
                    # select keys defined by columns from splitted line
                    split_line = line.strip().split()
                    values = [split_line[i] for i in indices]
                    print(src_file + ' ' + " ".join(values))


if __name__ == "__main__":
    main()

File: bin/printScoreFile_byHeader.pl
#!/usr/bin/perl

#index of $param1 in param2, with first index==0
sub indexArray{
 1 while $_[0] ne pop;
 $#_
}


#print the file $param1, and the columns labeled by @param2
sub printFile
  {
    my $fname=shift;
    open(INFILE, $fname) or die "couldn't open $fname\n";
    my $line;
    do {
	$line = <INFILE>;
    } while( ! ( $line =~ m/score/) );
    my @file_header=split(/\s+/, $line);
    my @indices;
    while($_=shift){
      push @indices, indexArray($_,@file_header);
    }
    my $line;
    while($line = <INFILE>){
      print "$fname ";
      chomp $line;  
      my @fields = split / +/,$line;
      for($i=0; $i <= $#indices ; $i++){
	print $fields[$indices[$i]]," ";
      }
      print "\n";
    }

  }

$file=shift;
# extract and print header
my @header;
print "src_file ";
while($_ = shift){
  push @header,$_;
  print "$_ ";
}
print "\n";
printFile($file,@header);

